sympy/assumptions/facts.py

```diff
@@ -53,40 +54,38 @@ def get_known_facts(x=None):
     """
     if x is None:
         x = Symbol('x')
+
     fact = And(
         # primitive predicates for extended real exclude each other.
-        # finite/infinite ones already reject each other so don't add here.
-        # primitive predicates exclude each other
-        Implies(Q.negative_infinite(x), ~Q.positive_infinite(x)),
-        Implies(Q.negative(x), ~Q.zero(x) & ~Q.positive(x)),
-        Implies(Q.positive(x), ~Q.zero(x)),
+        Exclusive(Q.negative_infinite(x), Q.negative(x), Q.zero(x),
+            Q.positive(x), Q.positive_infinite(x)),
 
-        # build real line and complex plane
-        Implies(Q.negative(x) | Q.zero(x) | Q.positive(x), ~Q.imaginary(x)),
-        Implies(Q.negative(x) | Q.zero(x) | Q.positive(x) | Q.imaginary(x), Q.algebraic(x) | Q.transcendental(x)),
+        # build complex plane
+        Exclusive(Q.real(x), Q.imaginary(x)),
+        Implies(Q.real(x) | Q.imaginary(x), Q.complex(x)),
 
         # other subsets of complex
-        Implies(Q.transcendental(x), ~Q.algebraic(x)),
-        Implies(Q.irrational(x), ~Q.rational(x)),
-        Equivalent(Q.rational(x) | Q.irrational(x), Q.negative(x) | Q.zero(x) | Q.positive(x)),
+        Exclusive(Q.transcendental(x), Q.algebraic(x)),
+        Equivalent(Q.real(x), Q.rational(x) | Q.irrational(x)),
+        Exclusive(Q.irrational(x), Q.rational(x)),
```

Comment:
Do we have `Equivalent(complex, transcental | algebraic)` anywhere?

Reply:
Not explicitly, but `complex` is always converted to `algebraic | transcendental` by predicate decomposition.

Reply:
I'll mark this as resolved since this approach is approved when predicate decomposition was introduced.